tests/python_package_test/test_engine.py

```diff
@@ -746,6 +749,23 @@ def test_ranking_prediction_early_stopping():
     with pytest.raises(AssertionError):
         np.testing.assert_allclose(ret_early, ret_early_more_strict)
 
+def test_ranking_with_position_information():
+    rank_example_dir = Path(__file__).absolute().parents[2] / 'examples' / 'lambdarank'
+    params = {
+        'objective': 'lambdarank',
+        'verbose': -1
+    }
+    copyfile(str(rank_example_dir / '_rank.train.position'), str(rank_example_dir / 'rank.train.position'))
+    lgb_train = lgb.Dataset(str(rank_example_dir / 'rank.train'), params=params)
+    gbm = lgb.train(params, lgb_train, num_boost_round=50)
+
+    with open(str(rank_example_dir / 'rank.train.position'), 'a') as file:
+        file.write('pos_1000')
+        file.close()
+    with np.testing.assert_raises_regex(lgb.basic.LightGBMError, "Positions size doesn't match data size"):
+        lgb.train(params, lgb_train, num_boost_round=50)
+
+    remove(str(rank_example_dir / 'rank.train.position'))
```

Comment:
Please don't use `os.remove()` or create files in the tests directory like this. Instead, please use the `tmp_path` fixture provided by `pytest` ([docs](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html)).

See here for an example:

[https://github.com/microsoft/LightGBM/blob/ac57d5a45a3ff3a04fbbcb8fb24f0c60072471e2/tests/python_package_test/test_basic.py#L139

https://github.com/microsoft/LightGBM/blob/ac57d5a45a3ff3a04fbbcb8fb24f0c60072471e2/tests/python_package_test/test_basic.py#L162-L163](https://github.com/microsoft/LightGBM/blob/ac57d5a45a3ff3a04fbbcb8fb24f0c60072471e2/tests/python_package_test/test_sklearn.py#L525-L529)

Reply:
Thank you! I addressed this.